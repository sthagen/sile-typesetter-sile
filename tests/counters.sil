\begin[papersize=a6]{document}
\nofolios
\use[module=packages.retrograde,target=v0.15.0]
\set[parameter=document.parindent,value=0pt]
\font[family=Libertinus Serif]
\use[module=packages.counters]
\set-counter[id=test,value=1234]

Default: \show-counter[id=test]

\set-counter[id=test,display=alpha]
Alpha: \show-counter[id=test]

\set-counter[id=test,display=ROMAN]
Roman: \show-counter[id=test]

\set-counter[id=test,display=greklow]
Greek: \font[language=el]{\show-counter[id=test]}

\set-counter[id=test,display=jpan]
Japanese: \font[family=Noto Sans CJK JP,language=ja]{\show-counter[id=test]}

\set-counter[id=test,display=arabext]
Arabic-Indic: \font[family=Noto Naskh Arabic,language=ar]{\show-counter[id=test]}

\end{document}
