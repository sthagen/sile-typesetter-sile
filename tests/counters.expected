Set paper size 	297.6377985	419.5275636
Begin page
Mx 	14.8819
My 	27.9564
Set font 	Libertinus Serif;10;400;;normal;;;LTR;
T	37 70 71 66 86 77 85 w=30.2600	(Default)
Mx 	45.1419
T	27 w=2.3600	(:)
Mx 	50.5286
T	18 19 20 21 w=18.6000	(1234)
Mx 	14.8819
My 	39.9564
T	34 77 81 73 66 w=24.7300	(Alpha)
Mx 	39.6119
T	27 w=2.3600	(:)
Mx 	45.0000
T	66 86 77 w=12.5200	(aul)
Mx 	14.8819
My 	51.9564
T	51 80 78 66 79 w=28.8000	(Roman)
Mx 	43.6819
T	27 w=2.3600	(:)
Mx 	49.0647
Mx 	8.3900
Mx 	6.4600
Mx 	6.4600
Mx 	6.6000
Mx 	6.6000
Mx 	6.6000
Mx 	3.2100
Mx 	6.5200
T	46 a=8.3900 36 a=6.4600 36 a=6.4600 57 a=6.6000 57 a=6.6000 57 a=6.6000 42 a=2.9700 55 a=6.5200	(MCCXXXIV)
Mx 	14.8819
My 	63.9564
Mx 	6.8500
Mx 	3.6400
Mx 	4.5400
Mx 	4.4700
Mx 	5.1200
T	40 a=6.8500 83 a=3.7200 70 a=4.4700 70 a=4.4700 76 a=5.1200	(Greek)
T	27 w=2.3600	(:)
Mx 	44.8883
T	811 w=1.9200	(͵)
Mx 	46.8083
T	859 877 869 862 w=21.5800	(ασλδ)
Mx 	68.3883
T	116 w=3.4100	(´)
Mx 	14.8819
My 	75.9564
T	43 66 81 66 79 70 84 70 w=35.8100	(Japanese)
Mx 	50.6919
T	27 w=2.3600	(:)
Mx 	56.0727
Set font 	Noto Sans CJK JP;10;400;;normal;;;LTR;
T	11657 w=10.0000	(千)
Mx 	66.1143
T	9669 w=10.0000	(二)
Mx 	76.1559
T	27692 w=10.0000	(百)
Mx 	86.1976
T	9492 w=10.0000	(三)
Mx 	96.2392
T	11655 w=10.0000	(十)
Mx 	106.2808
T	13131 w=10.0000	(四)
Mx 	14.8819
My 	87.9564
Set font 	Libertinus Serif;10;400;;normal;;;LTR;
T	34 83 66 67 74 68 w=27.1600	(Arabic)
Mx 	42.0419
T	14 w=3.3800	(-)
Mx 	45.4219
T	42 79 69 74 68 w=20.4400	(Indic)
Mx 	65.8619
T	27 w=2.3600	(:)
Mx 	71.2462
Set font 	Noto Naskh Arabic;10;400;;normal;;;LTR;
T	1268 1269 1270 1293 w=16.7200	(۱۲۳۴)
End page
Finish
